File: backend/app/schemas/observability/wakeups.py
from datetime import datetime

from app.schemas.common import ORMModel


class WakeupTaskOut(ORMModel):
    id: str
    target_type: str
    target_id: str
    target_name: str | None = None
    run_at: datetime
    reason: str | None = None
    status: str
    scheduled_by: str | None = None
    trigger_kind: str | None = None
    is_ai_wakeup: bool
    cancelled_at: datetime | None = None
    cancelled_reason: str | None = None
    created_at: datetime

File: frontend/src/api/types/wakeups.ts
export type WakeupTargetType = "cocoon" | "chat_group";

export type WakeupStatus = "queued" | "running" | "completed" | "failed" | "cancelled";

export interface WakeupTaskRead {
  id: string;
  target_type: WakeupTargetType;
  target_id: string;
  target_name: string | null;
  run_at: string;
  reason: string | null;
  status: WakeupStatus | string;
  scheduled_by: string | null;
  trigger_kind: string | null;
  is_ai_wakeup: boolean;
  cancelled_at: string | null;
  cancelled_reason: string | null;
  created_at: string;
}
